File: uri/uri_python/estruturas_e_bibliotecas/p2916.py
while True:
	try:
		entry = input().split(' ')
		N = int(entry[0])
		K = int(entry[1])
		entry = input().split(' ')
		for i in range(N):
			entry[i] = int(entry[i])
		entry.sort()
		soma = 0
		for i in range(K):
			soma += entry[N-1-i]
		print(soma)
	except EOFError:
		break

File: uri/uri_cpp/estruturas_e_bibliotecas/p2916.cpp
#include <iostream>
#include <sstream>
#include <string>
#include <stdlib.h>
#include <stdio.h>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    long long N, K, entry, i, soma;
    vector <long long> v;

    while (cin >> N) {
        cin >> K;
        for (i = 0; i < N; i++) {
            scanf("%ld", &entry);
            v.push_back(entry);
        }
        sort(v.begin(), v.end());
        soma = 0;
        for (i = N-1; i > (N-K-1); i--)
            soma += v[i];
        printf("%ld\n", (soma % 1000000007));
        v.clear();
    }

    return 0;
}